### PR TITLE
machines: Refactor SerialConsole for its transition to patternfly-react

### PR DESCRIPTION
**Pull request by mareklibra**
With this patch, the SerialConsole concern is splitted. From now on,
the component cares about cockpit channel (i.e. open, close, send/rcv) only,
while separating from the strictly front-end component which is being
transferred under the patternfly-react project [1].

Moving the compoment under external project [1] enables its reuse by other
projects like MageIQ or oVirt's VM Portal while sharing both the maintenance
and design.

[1] https://github.com/patternfly/patternfly-react
[2] https://github.com/xtermjs/xterm.js

---
 - [x] Depends on https://github.com/patternfly/patternfly-react/pull/160
 - [x] add `@patternfly/react-console` npm dependency once published

---

Future work (for reference):
- drop all `term.js-cockpit` references in favor of `<SerialConsole />` from `@patternfly/react-console`
- VncConsole

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -414,6 +414,7 @@ var aliases = {
     "d3": "d3/d3.js",
     "moment": "moment/moment.js",
     "react": "react-lite/dist/react-lite.js",
+    "react-dom": "react-lite/dist/react-lite.js",
```
Comment (larskarlitski) [inline webpack.config.js]: Are you using this anywhere?

File: package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "object-describer": "https://github.com/kubernetes-ui/object-describer.git#v1.0.4",
     "patternfly": "3.27.7",
     "patternfly-bootstrap-combobox": "1.1.7",
+    "patternfly-react": "0.24.0",
```
Comment (larskarlitski) [inline package.json]: I guess this is blocked until we have a release of `patternfly-react` which contains the `SerialConsole` component.

File: webpack.config.js (hunk 1 above)
Comment (mareklibra) [reply webpack.config.js]: Indirectly - the patternfly-react needs it.
The patternfly-react dist contains just the patternfly-react sources. Its dependencies are just listed in package.json and is up to consuming webpack to link it.

File: package.json (hunk 2 above)
Comment (mareklibra) [reply package.json]: Right. At least `0.25.0` will be required here, but let's see.

**Comment by mareklibra** [comment]
Rebase needed. Updated to keep track with patternfly-react PR - component moved to separate package.

**Comment by mareklibra** [comment]
Rebased.
Waiting for `@patternfly-react/console` to be published before adding it here.

**Comment by mareklibra** [comment]
Patternfly-react https://github.com/patternfly/patternfly-react/pull/160 PR merged, going on here.

**Comment by mareklibra** [comment]
patternfly-react dependency name changed to: `@patternfly/react-console`

**Comment by mareklibra** [comment]
This is ready for review.

**Comment by martinpitt** [comment]
This looks nice, thanks! Let's wait for tests, and I'll test this manually a bit. So this means we'll now include xterm.js, and can gather some experience with it. So I figure we should folllow suit with the terminals in docker, kubernetes, and the main "Terminal" menu too. @larskarlitski , are you aware of any downsides to this?

**Comment by larskarlitski** [comment]
> @larskarlitski , are you aware of any downsides to this?

No, I am not. Now that xterm.js 3 is stabilizing, it makes sense for us to have another look.

I'll try it out and give a review later. Thanks for your work @mareklibra!

**Comment by mareklibra** [comment]
Rebased.
The xterm v3 uses `<canvas>` instead of former `<div.terminal>`, so test updated accordingly.

**Pull request closed without merging**